src/client/Client.cc

```diff
@@ -10679,6 +10679,7 @@ int Client::_do_setxattr(Inode *in, const char *name, const void *value,
   req->head.args.setxattr.flags = xattr_flags;
 
   bufferlist bl;
+  assert (value != NULL || size != 0);
```

Comment:
@amitkumar50 "||" will cause it to not assert if value == null and size > 0, and we want it to assert in that case.

Reply:
@batrick `assert(value || size == 0)` Means we will assert when size != 0 ie if size>0 we need to assert?